+adi/+AD4020/Rx.m

```diff
@@ -0,0 +1,111 @@
+classdef Rx < adi.common.Rx & matlabshared.libiio.base & adi.common.Attribute
+    % AD4020 Precision ADC Class
+    % adi.AD4020.Rx Receives data from the AD4020 ADC
+    %   The adi.AD4020.Rx System object is a signal source that can receive
+    %   data from the AD4020.
+    %
+    %   rx = adi.AD4020.Rx;
+    %   rx = adi.AD4020.Rx('uri','192.168.2.1');
+    %
+    %   <a href="https://www.analog.com/media/en/technical-documentation/data-sheets/ad4020-4021-4022.pdf">AD4020 Datasheet</a>
+
+    properties (Nontunable)
+        % SamplesPerFrame Samples Per Frame
+        %   Number of samples per frame, specified as an even positive
+        %   integer.
+        SamplesPerFrame = 4096
+    end
+
+    properties (Dependent)
+        % Voltage
+        %   ADC Voltage in mV
+        Voltage
+
+        % VoltageScale Voltage Scale
+        %   ADC Voltage scale.
+        VoltageScale
+
+        % VoltageOffset Voltage Offset
+        %   ADC Voltage offset.
+        VoltageOffset
+    end
+
+    properties
+        % SamplingFrequency
+        %   Sampling Frequency in Hertz.
+        SamplingFrequency = 0
```

Comment:
Should the sampling frequency be set to 0? Or should it be 1800000 (1.8 MSPS)?

Reply:
SamplingFrequency property value is actually updated by the setupInit() method once connected to the device.
isn't this the way to make use of it ?

Reply:
Oh I just noticed that the `setupInit` method is calling `getDeviceAttributeLongLong` (as opposed to `setDeviceAttributeLongLong`)

Is the sampling_frequency attribute configurable in the Linux driver? If it is read-only, then I guess it makes sense to replace 0 with whatever value is set in the driver, and remove the call to `getDeviceAttributeLongLong` (as it doesn't accomplish anything)

If it is configurable in the Linux driver, then we should add a setter method for the SamplingFrequency property to allow the user to choose a different value than the default, and also replace `getDeviceAttributeLongLong` with `setDeviceAttributeLongLong` in the `setupInit` method, so that the value is applied in the hardware/driver whenever connection is established. Something like this: https://github.com/analogdevicesinc/PrecisionToolbox/blob/main/%2Badi/%2BAD7768/Rx.m#L74-L101 . I guess @tfcollins mentioned something to this effect here: https://github.com/analogdevicesinc/PrecisionToolbox/pull/11#discussion_r1336422179

Do you have any suggestions/recommendations here @tfcollins ?

Reply:
If there is a defined set you should use an Enum or the mustBeMember option like: https://github.com/analogdevicesinc/RFMicrowaveToolbox/blob/main/%2Badi/%2Binternal/XUD1aControl.m#L22

Reply:
> Oh I just noticed that the `setupInit` method is calling `getDeviceAttributeLongLong` (as opposed to `setDeviceAttributeLongLong`)
> 
> Is the sampling_frequency attribute configurable in the Linux driver? If it is read-only, then I guess it makes sense to replace 0 with whatever value is set in the driver, and remove the call to `getDeviceAttributeLongLong` (as it doesn't accomplish anything)

The sampling_frequency attribute is read-only at the moment indeed.
Will remove the property altogether to avoid any confusion and simplify the code (as it doesn't accomplish anything like you said).